### DISCUSSION TITLE
Error while obtaining sizes from MEX S-function 'casadi_fun'

### DISCUSSION THREAD

**Discussion by MSadiq89** [casadi/casadi#3202] [Q&A]
![casadi_Help1](https://github.com/casadi/casadi/assets/22737377/23f82895-6a7a-42a7-af30-07c631a39176)

**Comment by MSadiq89** [comment]
I am trying to generate code on a dspace target (using configuration desk software) but getting this error. The code generation seems to work fine for generic computer code generation properties.

**Comment by jgillis** [comment]
It looks like `mdlInitializeSizes` is being called but something goes wrong loading 'f.casadi'. Any options to increase verbosity of error messages in Simulink?

I'm not experienced in D-Space, but I would guess that the usage of saved casadi files like `f.casadi` is anyhow not supported.
You probably need to do proper casadi side code-generation cfr https://github.com/casadi/casadi/discussions/3202

**Reply by MSadiq89** [reply]
Thankyou for the response. I contacted Dspace support, here's what they have to say regarding it

Dear Mr. Sadiq,

Many thanks for contacting the dSPACE Support.

What are you facing is a Simulink Code generation error indicating there is something missing or incompatible. Most probably because your MEX file has dependencies that are not available (made defined) in the ConfigurationDesk environment. Have you contacted CasADi about these dependencies? 
Maybe also due to outdated MEX DLL. I would recreate the MEX DLL of this S-Function anyway. 

Please make sure in all cases that the S-Function and its related MEX DLL are available/defined to the ConfigurationDesk environment?
For this, consider our FAQ 298 for S-Function Handling: http://www.dspace.com/faq?298

Please also make sure that no security tool is running during the ConfigurationDesk build, and you have enough user access rights from within ConfigurationDesk?


Please report the result for:
- Update Model Ctrl-D?
- dsrt_build('system', <modelName>, 'GenerateSIC', false)?
- dsrt_build('system', <modelName>, 'GenerateSIC', true) ?
- Change the system target of your model to grt.tlc and apply Ctrl-B on the model (Code generation for the MATLAB target)?

**Comment by jgillis** [comment]
Yeah so working with `f.casadi` brings in a dependency on the casadi runtime (casadi.dll), while casadi codegen results in a strategy without dependencies.
The linked document provides a possible path to achieve compatibility of ConfigurationDesk and casadi.dll (it talks about cpp source file), but that is going to take quite some manhours to implement

**Reply by MSadiq89** [reply]
Alright, if i am getting it right, do we need to get IPOPT source code in a S-function in simulink?

Could you kindly guide the process of achieving the solution.

**Reply by jgillis** [reply]
My suggestion is to *not* use ipopt and use casadi codegen.
Getting IPOPT running on Dspace interests me, but I have no funding to support any work in that direction
